### DISCUSSION TITLE
False difference detection when using "comments"?

### DISCUSSION THREAD

**Discussion by mgzenitech** [prantlf/jsonlint#30] [General]
Test case:

`jsonlint.json`:
```json
{
  "check": true,
  "comments": true,
  "continue": true,
  "enforce-double-quotes": true,
  "no-duplicate-keys": true,
  "sort-keys": true,
  "trailing-newline": true,
  "trim-trailing-commas": true
}
```

`test.json`
```jsonc
{
  "a": "b" //aaaa
}
```

test run:
```sh
jsonlint -f jsonlint.json test.json
```

```sh
File: test.json
1 hunk differs
===================================================================
--- test.json.orig
+++ test.json
@@ -1,3 +1,3 @@
 {
-  "a": "b" //aaaa
+  "a": "b"
 }
```

I'm a missing something in config?

**Comment by prantlf** [comment]
Yes, you are, but this one is tricky :-) TL;DR: Add this to `jsonlint.json`:

```
  "pretty-print": true
```

Options `comments` and `trailing-commas` make the parser ignore those two and they won't be available when the output is printed _by default_. If you want to retain comments or trailing commas in the output, you have to add the option `pretty-print`. It will try to retain also these two, which otherwise have no value in the parsed JSON.

Also, depending on your preference, you should set `trailing-newline` to `true` or `false` to enforce the standard and avoid false difference reports.
